data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultReactiveCosmosRepositoryOperations.java

```diff
@@ -0,0 +1,712 @@
+            boolean dtoProjection = preparedQuery.isDtoProjection();
+            boolean isEntity = preparedQuery.getResultDataType() == DataType.ENTITY;
+            if (isEntity || dtoProjection) {
+                return findOneEntityOrDto(preparedQuery, paramList);
+            } else {
+                return findOneCustomResult(preparedQuery, paramList);
+            }
+        } catch (CosmosException e) {
+            if (e.getStatusCode() == HttpResponseStatus.NOT_FOUND.code()) {
+                return Mono.empty();
+            }
+            return Mono.error(e);
+        }
+    }
+
+    @Override
+    @NonNull
+    public <T> Mono<T> findOptional(@NonNull Class<T> type, @NonNull Serializable id) {
+        return findOne(type, id).onErrorReturn(EmptyResultException.class, (T) Mono.empty());
+    }
+
+    @Override
+    @NonNull
+    public <T, R> Mono<R> findOptional(@NonNull PreparedQuery<T, R> preparedQuery) {
+        return findOne(preparedQuery).onErrorReturn(EmptyResultException.class, (R) Mono.empty());
+    }
+
+    @Override
+    @NonNull
+    public <T> Flux<T> findAll(PagedQuery<T> pagedQuery) {
+        throw new UnsupportedOperationException("The findAll method without an explicit query is not supported. Use findAll(PreparedQuery) instead");
+    }
+
+    @Override
+    @NonNull
+    public <T> Mono<Long> count(PagedQuery<T> pagedQuery) {
+        throw new UnsupportedOperationException("The count method without an explicit query is not supported. Use findAll(PreparedQuery) instead");
+    }
+
+    @Override
+    @NonNull
+    public <T, R> Flux<R> findAll(@NonNull PreparedQuery<T, R> preparedQuery) {
+        try {
+            boolean dtoProjection = preparedQuery.isDtoProjection();
+            boolean isEntity = preparedQuery.getResultDataType() == DataType.ENTITY;
+            List<SqlParameter> paramList = new ParameterBinder().bindParameters(preparedQuery);
+            if (isEntity || dtoProjection) {
+                Argument<R> argument;
+                if (dtoProjection) {
+                    argument = Argument.of(ReflectionUtils.getWrapperType(preparedQuery.getResultType()));
+                } else {
+                    argument = Argument.of(preparedQuery.getResultType());
+                }
+                CosmosPagedFlux<ObjectNode> result = getCosmosResults(preparedQuery, paramList, ObjectNode.class);
+                return result.map(item -> deserialize(item, argument));
+            } else {
+                DataType dataType = preparedQuery.getResultDataType();
+                Class<R> resultType = preparedQuery.getResultType();
+                CosmosPagedFlux<?> result = getCosmosResults(preparedQuery, paramList, getDataTypeClass(dataType));
+                return result.mapNotNull(item -> {
+                    if (resultType.isInstance(item)) {
+                        return (R) item;
+                    } else if (item != null) {
```

Comment:
Another not needed `else`

Reply:
Changed https://github.com/micronaut-projects/micronaut-data/pull/1797/commits/8242884205f015b9a077bd3cc9a1318f31c4bdca